src/utils/internalLinks.js

```diff
@@ -20,16 +22,18 @@ const getPathsFromUrl = (url: string = '', realm: string) => {
 };
 
 /** PRIVATE -- exported only for tests. */
-export const isInternalLink = (url: string, realm: string): boolean =>
-  isUrlOnRealm(url, realm) ? /^(\/#narrow|#narrow)/i.test(url.split(realm).pop()) : false;
+export const isInternalLink = (url: string, realm: URL): boolean =>
+  isUrlOnRealm(url, realm)
+    ? /^(\/#narrow|#narrow)/i.test(url.split(realm.toString()).pop())
```

Comment:
This bit deserves a TODO comment about replacing it with `URL`, similar to a couple of other such comments added in this branch.